searchlib/src/main/java/com/yahoo/searchlib/aggregation/Group.java

```diff
@@ -12,23 +12,45 @@
 import com.yahoo.vespa.objects.Serializer;
 
 import java.util.ArrayList;
-import java.util.Collections;
+import java.util.Comparator;
 import java.util.Iterator;
 import java.util.List;
+import java.util.stream.Collectors;
 
 public class Group extends Identifiable {
 
     public static final int classId = registerClass(0x4000 + 90, Group.class);
     private static final ObjectPredicate REF_LOCATOR = new RefLocator();
-    private List<Integer> orderByIdx = new ArrayList<>();
-    private List<ExpressionNode> orderByExp = new ArrayList<>();
-    private List<AggregationResult> aggregationResults = new ArrayList<>();
-    private List<Group> children = new ArrayList<>();
+    private List<Integer> orderByIdx = List.of();
+    private List<ExpressionNode> orderByExp = List.of();
+    private List<AggregationResult> aggregationResults = List.of();
+    private List<Group> children = List.of();
     private ResultNode id = null;
     private double rank;
     private int tag = -1;
     private SortType sortType = SortType.UNSORTED;
 
+    private static <T> List<T> add(List<T> oldList, T obj) {
+        if (oldList.isEmpty()) {
+            return List.of(obj);
+        }
+        if (oldList.size() == 1) {
+            return List.of(oldList.get(0), obj);
+        }
+        List<T> newList = (oldList instanceof ArrayList) ? oldList : new ArrayList<>(oldList);
+        newList.add(obj);
+        return newList;
+    }
+
+    private static <T> List<T> sort(List<T> list, Comparator<T> cmp) {
+        if (list instanceof ArrayList) {
+            list.sort(cmp);
+            return list;
+        } else {
+            return list.stream().sorted(cmp).collect(Collectors.toList());
```

Comment:
I do not think this will work. The reason for this helper was to be able to sort when the list is unmodifiable too. That will be the case when it has very few elements.

Reply:
No need to return modifiable list. But need to handle longer than 2, and not ArrayList to avoid code coupling that is not obvious.